[DCP196354] ExtraDocsFieldDescriptors (#2850)
SkylineCommunications/dataminer-docs: user-guide/Advanced_Modules/DOM/DOM_objects/DOM_FieldDescriptors/DOM_DomInstanceFieldDescriptor.md

---
uid: DOM_DomInstanceFieldDescriptor
---

# DomInstanceFieldDescriptor

> - **FieldValue type**: `Guid` of a `DOMInstance`

Reviewer: ```suggestion
- **FieldValue type**: `Guid`
```

- **Multiple values supported**: :heavy_check_mark:

| Type of Descriptor | FieldType | FieldValue type |
|--------------------|-----------|-----------------|

> | DOMInstance | Guid | Guid |

Reviewer: ```suggestion
| References a single `DomInstance` | Guid | Guid |
```

> | DOMInstance with multiple values enabled| List\<Guid\> | Guid (ListValueWrapper) |

Reviewer: ```suggestion
| References one or more `DomInstances` | List\<Guid\> | Guid (ListValueWrapper) |
```

> Available from DataMiner 10.1.10/10.2.0 onwards. Can be used to define that a field should contain the `Guid` of a `DomInstance`. This `DomInstance` can exist in a different DOM manager. This is why the descriptor has a `ModuleId` property that defines where the instances can be found. There is also a `DomDefinitionIds` list property that can be used to define whether `DomInstances` should be linked to the defined definitions. Both properties are intended for UIs, and their validity and existence is not checked server-side. The `Guid` of the selected `DomInstance` is saved.

Reviewer: ```suggestion
Available from DataMiner 10.1.10/10.2.0 onwards. Can be used to define that a field should contain the ID of a `DomInstance` in the form of a `Guid`. This `DomInstance` can exist in a different DOM manager. This is why the descriptor has a `ModuleId` property that defines where the instances can be found. There is also a `DomDefinitionIds` list property that can be used to define whether `DomInstances` should be linked to the defined definitions. Both properties are intended for UIs, and their validity and existence is not checked server-side.
```

To enable multiple values, set the `FieldType` to `List<Guid>`.

```csharp
var descriptor = new DomInstanceFieldDescriptor(domManagerModuleId)
{
    ID = new FieldDescriptorID(Guid.NewGuid()),
    Name = "DomInstanceFieldDescriptor",

>     DomDefinitionIds = { domDefinitionGuid },

Reviewer: ```suggestion
    DomDefinitionIds = { domDefinitionId },
```

    FieldType = typeof(Guid)
};
```

Assigning a `FieldValue` to the `FieldDescriptor` of a new `DomInstance`:

```csharp
var instance = new DomInstance 
{        
    ID = new DomInstanceId(Guid.NewGuid()),
    DomDefinitionId = domDefinitionId
};

> instance.AddOrUpdateFieldValue(sectionDefinition, descriptor, GuidOfDomInstance); // type should be Guid

Reviewer: ```suggestion
var otherInstanceId = Guid.Parse("b17e18bd-ab2d-4ed8-80bc-4098140fe6da");
instance.AddOrUpdateFieldValue(sectionDefinition, descriptor, otherInstanceId); // Type should be Guid
```

> // example with multiple values enabled 

Reviewer: ```suggestion
// Example with multiple values enabled 
```

> instance.AddOrUpdateFieldValue(sectionDefinition, descriptor, new ListValueWrapper<Guid>(Guid1, Guid2));

Reviewer: ```suggestion
var firstInstanceId = Guid.Parse("755a424e-783f-466d-981e-8359fd0ca426");
var secondInstanceId = Guid.Parse("16da2b5d-6b43-4267-96ba-4abd5c16ee2b");
instance.AddOrUpdateFieldValue(sectionDefinition, descriptor, new ListValueWrapper<Guid>(firstInstanceId, secondInstanceId));
```
